server/plugin/plugin.go

```diff
@@ -271,10 +268,51 @@ func (p *Plugin) OnActivate() error {
 	return nil
 }
 
+// Initialize telemetry setups the tracker/clients needed to send telemetry data.
+// - enable/disable is handled by LogSettings.EnableDiagnostics (also at onConfigurationChange for live changes)
+// - debug logging is added at pluginAPI side if ServiceSettings.EnableDeveloper is true
+func (p *Plugin) initializeTelemetry() {
+	var err error
+	var trackerLogger logger.Logger
+
+	// Telemetry client
+	p.telemetryClient, err = telemetry.NewRudderClient()
+	if err != nil {
+		p.API.LogWarn("Telemetry client not started", "error", err.Error())
+		return
+	}
+
+	// Get config values
+	enableDiagnostics := false
+	enableDeveloper := false
+	if config := p.API.GetConfig(); config != nil {
+		if configValue := config.LogSettings.EnableDiagnostics; configValue != nil {
+			enableDiagnostics = *configValue
+		}
+		if configValue := config.ServiceSettings.EnableDeveloper; configValue != nil {
+			enableDeveloper = *configValue
+		}
+	}
+	if enableDeveloper {
+		trackerLogger = logger.New(p.API)
+	}
```

Comment:
It's a bit of a bummer that the logger doesn't get updated when `EnableDeveloper` changes. Maybe it's a  `SetLogger` helps here, even though it's quite Java-like.

Reply:
oh, that's a really nice catch @hanzei 

SetLogger does the job but it's a bit surprising for a `Tracker` interface.
I'm thinking of several options:

SetLogger -> verbose and explicit but hard to see the relation between Tracker and SetLogger
```go
type Tracker interface {

	TrackEvent(event string, properties map[string]interface{}) error
	TrackUserEvent(event string, userID string, properties map[string]interface{}) error
	Enable()
	Disable()
        SetLogger(l logger.Logger)
}
```


merge enable/disable/setlogger into ReloadConfig -> less API surface, hides complexity, less verbose but also less semantic 
```go
type Tracker interface {
	TrackEvent(event string, properties map[string]interface{}) error
	TrackUserEvent(event string, userID string, properties map[string]interface{}) error
        ReloadConfig(enableDiagnostic, enableDeveloper bool)
}
```

same but with a new type for config 
```go
type TrackerConfig struct {
    enableDiagnostics bool
    enableLogger bool // could be enableDeveloper, 0/5 still on that
}

type Tracker interface {
	TrackEvent(event string, properties map[string]interface{}) error
	TrackUserEvent(event string, userID string, properties map[string]interface{}) error
        ReloadConfig(config TrackerConfig)
}
```

I'm more inclined to use the last one, thoughts?

Reply:
I really like the second approach (with the new type for config, that can be enriched with additional stuff in the future potentially).

Reply:
updated code to follow this approach